src/layout.rs

```diff
@@ -335,6 +336,26 @@ impl TapHoldTracker {
     }
 }
 
+#[derive(Debug)]
+struct TriStateLayer {
+    activation_layers: (usize, usize),
+    target_layer: usize,
+}
+
+impl TriStateLayer {
+    fn apply(&self, last_layer: usize, second_to_last_layer: usize) -> Option<usize> {
+        if (last_layer == self.activation_layers.0
+            && second_to_last_layer == self.activation_layers.1)
+            || (last_layer == self.activation_layers.1
+                && second_to_last_layer == self.activation_layers.0)
+        {
+            Some(self.target_layer)
+        } else {
+            None
+        }
+    }
+}
```

Comment:
This is fine, but maybe a match statement would be slightly neater?

Reply:
Maybe the following is a little more compact:
```
    fn apply(&self, layer_0: usize, layer_1: usize) -> Option<usize> {
        (self.activation_layers == (layer_0, layer_1)
            || self.activation_layers == (layer_1, layer_0))
            .then_some(self.target_layer)
    }

```

What do you think?

Reply:
Awesome